File: itlen/app/src/main/java/com/deadely/itl_en/di/module/AppModule.java
package com.deadely.itl_en.di.module;

import com.deadely.itl_en.App;

import dagger.Module;

@Module
public class AppModule {
    private App app;

    public AppModule(App app) {
        this.app = app;
    }

    public App getApp() {
        return app;
    }
}

File: itlen/app/src/main/java/com/deadely/itl_en/di/module/AppModule.kt
package com.deadely.itl_en.di.module

import android.app.Application
import com.deadely.itl_en.App
import com.deadely.itl_en.di.scope.PerApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val app: App) {
    @Provides
    @Singleton
    @PerApplication
    fun provideApplication(): Application {
        return app
    }


}